### ISSUE TITLE
Dimension mismatch with loading a lazily loaded image to memory

### ISSUE THREAD

**Issue by RpfR2000** [tlnagy/TiffImages.jl#177]
I have loaded in a large image (26 GB) lazily with the lazyio=true flag and am now trying to load a subset of that image to memory by indexing as follows: 

```          
chunk = Array{Gray{N0f16}, 3}(undef, y_end-y_start+1, x_end-x_start+1, slices)
for k in 1:slices
    chunk[:,:,k] .= image[y_start:y_end, x_start:x_end, k]
end
```

This gives the following error: LoadError: DimensionMismatch: parent has 18596 elements, which is incompatible with size (256, 256). The reason I am loading in a loop like this is because I couldn't even @show the size of a 3D chunk of the image without the same error being thrown. I can @show the size of a slice, though (not sure why). I have verified that the size and eltype of the lazily loaded image is correct. Any ideas what is going on?

**Comment by tlnagy** [comment]
Does `image[y_start:y_end, x_start:x_end, k]` give you the dimensions and eltype of what you expect? Have you tried loading the image with `mmap=true` instead of lazyio?

**Comment by RpfR2000** [comment]
Yes, `image[y_start:y_end, x_start:x_end, k]` gives the correct eltype and dimensions. I just tried mmap=true and got LoadError: BoundsError: attempt to access 0-element Vector{TiffImages.Tag} at index [1].

**Comment by tlnagy** [comment]
Interesting. Can you give me the output of `first(ifds(img))`?

**Comment by RpfR2000** [comment]
first(ifds(image)) = IFD, with tags:                  
        Tag(SUBFILETYPE, 2)                           
        Tag(IMAGEWIDTH, 10138)                        
        Tag(IMAGELENGTH, 10138)                       
        Tag(BITSPERSAMPLE, 16)                        
        Tag(COMPRESSION, COMPRESSION_NONE)            
        Tag(PHOTOMETRIC, 1)                           
        Tag(IMAGEDESCRIPTION, "<?xml version="1.0" ...")
        Tag(SAMPLESPERPIXEL, 1)                       
        Tag(SOFTWARE, "OME Files (C++) 0.6....")      
        Tag(TILEWIDTH, 256)                           
        Tag(TILELENGTH, 256)                          
        Tag(TILEOFFSETS, UInt64[16, 131088, 262160, 393232, 524304, ...])
        Tag(TILEBYTECOUNTS, UInt64[131072, 131072, 131072, 131072, 131072, ...])
        Tag(SAMPLEFORMAT, 1)                          
        Tag(UNKNOWN(65325), 1329.7020542975515)       
        Tag(UNKNOWN(65326), 0.065)                    
        Tag(UNKNOWN(65327), 1.0)                      
        Tag(UNKNOWN(65328), 1.0)                      
        Tag(UNKNOWN(65329), Float64[-13174.5, -14112.900000000001, 4663.08, 1329.7020542975515])

**Comment by tlnagy** [comment]
Ah, the image is tiled. Both `mmap` and `lazyio` don't support tiled images if I remember properly.

> This gives the following error: LoadError: DimensionMismatch: parent has 18596 elements, which is incompatible with size (256, 256).

This error now makes more sense, since the tile sizes are 256x256. Do you mind providing the full stacktrace (all of the output) for `lazyio=true` error? I'm curious where it's failing.

**Comment by RpfR2000** [comment]
size(img.cache) doesn't work, I get `type DenseTaggedImage has no field cache`. 

`size(image) = (10138, 10138, 135)`

Full stacktrace:
```
ERROR: LoadError: DimensionMismatch: parent has 18596 elements, which is incompatible with size (256, 256)
Stacktrace:
  [1] _throw_dmrs(n::Int64, str::String, dims::Tuple{Int64, Int64})
    @ Base ./reshapedarray.jl:182
  [2] _reshape
    @ ./reshapedarray.jl:177 [inlined]
  [3] reshape
    @ ./reshapedarray.jl:112 [inlined]
  [4] reshape
    @ ./reshapedarray.jl:117 [inlined]
  [5] _broadcast_getindex_evalf
    @ ./broadcast.jl:709 [inlined]
  [6] _broadcast_getindex
    @ ./broadcast.jl:682 [inlined]
  [7] _getindex
    @ ./broadcast.jl:706 [inlined]
  [8] _broadcast_getindex
    @ ./broadcast.jl:681 [inlined]
  [9] getindex
    @ ./broadcast.jl:636 [inlined]
 [10] macro expansion
    @ ./broadcast.jl:1004 [inlined]
 [11] macro expansion
    @ ./simdloop.jl:77 [inlined]
 [12] copyto!
    @ ./broadcast.jl:1003 [inlined]
 [13] copyto!
    @ ./broadcast.jl:956 [inlined]
 [14] copy
    @ ./broadcast.jl:928 [inlined]
 [15] materialize
    @ ./broadcast.jl:903 [inlined]
 [16] tile(cache::Matrix{Gray{N0f16}}, ifd::TiffImages.IFD{UInt64})
    @ TiffImages ~/.julia/packages/TiffImages/Xrmi4/src/load.jl:151
 [17] transform
    @ ~/.julia/packages/TiffImages/Xrmi4/src/load.jl:139 [inlined]
 [18] getindex(A::LazyBufferedTIFF{Gray{N0f16}, UInt64, Matrix{Gray{N0f16}}}, i1::Int64, i2::Int64, i::Int64)
    @ TiffImages ~/.julia/packages/TiffImages/Xrmi4/src/types/lazy.jl:129
 [19] macro expansion
    @ ./multidimensional.jl:917 [inlined]
 [20] macro expansion
    @ ./cartesian.jl:64 [inlined]
 [21] macro expansion
    @ ./multidimensional.jl:912 [inlined]
 [22] _unsafe_getindex!
    @ ./multidimensional.jl:925 [inlined]
 [23] _unsafe_getindex(::IndexCartesian, ::LazyBufferedTIFF{Gray{N0f16}, UInt64, Matrix{Gray{N0f16}}}, ::UnitRange{Int64}, ::UnitRange{Int64}, ::Int64)
    @ Base ./multidimensional.jl:903
 [24] _getindex
    @ ./multidimensional.jl:889 [inlined]
 [25] getindex
    @ ./abstractarray.jl:1291 [inlined]
 [26] getindex
    @ ~/.julia/packages/TiffImages/Xrmi4/src/types/dense.jl:67 [inlined]
```

**Comment by tlnagy** [comment]
Ah, my bad. I forgot that we wrap with a `DenseTaggedImage`, could you give me the output of `size(img.data.cache)`?

**Comment by RpfR2000** [comment]
```
size(image.data.cache) = (10240, 10240)
```

**Comment by RpfR2000** [comment]
@tlnagy perhaps I could also share the image with you if that would be easier? It is quite large though.

**Comment by tlnagy** [comment]
Yeah, do you mind uploading it to dropbox/similar place and I can download it? It'll be easier to diagnose that way then going back and forth here.

**Comment by RpfR2000** [comment]
Sent

**Comment by tlnagy** [comment]
Do you mind sharing it as a public link? It's requiring me to log in to access it.

**Comment by RpfR2000** [comment]
Sorry, just sent a link. Let me know if that works.

**Comment by chrstphrbrns** [comment]
I'm preparing a PR for this

**Comment by RpfR2000** [comment]
Does this require a PR?

**Comment by tlnagy** [comment]
Thanks to @chrstphrbrns' quick work this should be fixed in the newest release `v0.11.2`, give it a hour or so for the version to propagate through CI etc.
